packages/box/src/Box.spec.tsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import '@testing-library/jest-dom/extend-expect';
+import { render, cleanup } from '@testing-library/react';
+import Box from '.';
+import { BoxProps } from './Box';
+
+// types
+interface SharedProps {
+  name?: string;
+}
+
+interface RenderedEls {
+  box?: HTMLElement | null;
+  child?: HTMLElement | null;
+}
+
+interface LinkWrapperProps {
+  href: string;
+  target: string;
+  children: React.ReactNode;
+}
+
+type testProps = BoxProps<SharedProps>;
+
+describe('packages/box', () => {
+  const renderedEls: RenderedEls = {};
+  const linkWrapperFn = jest.fn();
+
+  function LinkWrapper({ href, target, children, ...rest }: LinkWrapperProps) {
+    linkWrapperFn();
+    return (
+      <span {...rest}>
+        <a data-testid="link-component" href={href} target={target}>
+          {children}
+        </a>
+      </span>
+    );
+  }
+
+  const sharedProps = { name: 'testName' };
+  const anchorProps = {
+    href: 'https://cloud.mongodb.com',
+    target: '_blank',
+    ...sharedProps,
+  };
+  const linkProps = { component: LinkWrapper, ...anchorProps };
+
+  const renderBox = (props: testProps = sharedProps) => {
+    const { queryByTestId } = render(
+      <Box data-testid="box" {...props}>
+        <div data-testid="child">Child Content</div>
+      </Box>,
+    );
+
+    renderedEls.child = queryByTestId('child');
+    renderedEls.box = queryByTestId('box');
+  };
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    cleanup();
```

Comment:
OH! I didn't realize `cleanup` actually takes care of the unmounting for you. Good improvement, I should fix that in my previous tests! I can't believe I missed that in the docs.